### PR TITLE
feat: Add an example e2e test environment / e2e tests for GRPC service.

### PR DESCRIPTION
**Pull request by agarwalnit**
**Description**
This PR adds and example GRPC service (visitor service ) with a database which stores visit counts for an user.  It provides an example on how to use skaffold/ minikube to shape development experience by starting a local E2E environment and execute tests against it. 

The end to end environment provide an example to start the service locally, with a spanner emulator and a database bootsrap script.  It further provides E2E tests which can be executed against the local E2E environment.

### CODE REVIEW

File: examples/grpc-e2e-tests/tests/go.mod [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module github.com/GoogleContainerTools/skaffold/examples/grpc-e2e-tests/tests
+
+go 1.18
```
Comment (tejal29) [inline examples/grpc-e2e-tests/tests/go.mod:3]: Please move to v1.19 due to current CVE in go.18

File: examples/grpc-e2e-tests/service/go.mod [hunk 2]
```diff
@@ -0,0 +1,34 @@
+module github.com/GoogleContainerTools/skaffold/examples/grpc-e2e-tests/service
+
+go 1.18
```
Comment (tejal29) [inline examples/grpc-e2e-tests/service/go.mod:3]: go 1.19 due to recent CVEs

**Review by tejal29 — COMMENTED** [review]
Thank you so much for this example @agarwalnit. 

Two comments:
1) Can you please move this example from `examples` to `integration/examples` ?
2) Can you please move to go v1.19?  - Not a big deal. 

Please let me know if you need help. Sorry for the delay in the review.

File: examples/grpc-e2e-tests/tests/go.mod (hunk 1 above)
Comment (agarwalnit) [reply examples/grpc-e2e-tests/tests/go.mod]: Thanks , moved to 1.19.

File: examples/grpc-e2e-tests/service/go.mod (hunk 2 above)
Comment (agarwalnit) [reply examples/grpc-e2e-tests/service/go.mod]: Thanks, moved to 1.19

**Comment by agarwalnit** [comment]
> Thank you so much for this example @agarwalnit.
> 
> Two comments:
> 
> 1. Can you please move this example from `examples` to `integration/examples` ?
> 2. Can you please move to go v1.19?  - Not a big deal.
> 
> Please let me know if you need help. Sorry for the delay in the review.

Thanks for doing the review for this very large CL. Hopefully, it provide an example for to create a local E2E environment and related tests for users working on similar use cases. Thanks moved all code to integration/examples. Also moved to go 1.19.

**Comment by aaron-prindle** [comment]
@agarwalnit can you rebase I resubmit?  I believe we are seeing some issues that have since been resolved @ HEAD.  I you can rebase from the latest `main` branch I think we can get all green checks on our CI/CD

**Comment by agarwalnit** [comment]
> Contributor

Thanks, rebased changes with the head on skaffold.

File: integration/examples/grpc-e2e-tests/skaffold.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+apiVersion: skaffold/v2beta29
```
Comment (aaron-prindle) [inline integration/examples/grpc-e2e-tests/skaffold.yaml:1]: Can you change this to be skaffold/v3?

**Comment by aaron-prindle** [comment]
I think currently your PR is carrying some additional changes not related to the new example, can you remove the other changes that are not related to `integration/examples/grpc-e2e-tests/*`.  Currently there are ~476 files that are touched I think only the files in `integration/examples/grpc-e2e-tests/*` should be in the PR

Also can you change the apiVersion for the example's `skaffold.yaml` to be `skaffold/v3` (from `skaffold/v2beta29`).  This is currently causing a `lint` test failure in our CI/CD

**Comment by agarwalnit** [comment]
> I think currently your PR is carrying some additional changes not related to the new example, can you remove the other changes that are not related to `integration/examples/grpc-e2e-tests/*`. Currently there are ~476 files that are touched I think only the files in `integration/examples/grpc-e2e-tests/*` should be in the PR
> 
> Also can you change the apiVersion for the example's `skaffold.yaml` to be `skaffold/v3` (from `skaffold/v2beta29`). This is currently causing a `lint` test failure in our CI/CD

Aha, I think I executed rebase in a wrong manner. Fixed this and also upgraded skaffold version.

**Review by aaron-prindle — APPROVED** [review]
LGTM, waiting on green CI/CD tests and we can merge

**Pull request merged**